web/src/components/browse/namespace-long-row.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useRef, useEffect, useCallback } from 'react';
+
+import { BiggestNamespaceResults } from '../../../types';
+
+type Props = {
+  namespaces: BiggestNamespaceResults[] | undefined;
+  selectedNamespace: string | undefined;
+  handleSelectNamespace: (selectedNamespace: string) => void;
+};
+
+export const NamespaceLongRow = (props: Props) => {
+  const {
+    namespaces,
+    selectedNamespace,
+    handleSelectNamespace
+  } = props;
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const itemRefs = useRef<{ [key: string]: HTMLDivElement | null }>({});
+
+  const isInViewport = (element: HTMLElement): boolean => {
+    const rect = element.getBoundingClientRect();
+    const padding = 12; // Adjust this value to increase or decrease the padding
+
+    return (
+      rect.top >= 0 - padding &&
+      rect.left >= 0 - padding &&
+      rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) + padding &&
+      rect.right <= (window.innerWidth || document.documentElement.clientWidth) + padding
+    );
+  };
+
+  const scrollToItem = useCallback((namespace: string) => {
+    const element = itemRefs.current[namespace];
+    if (element && !isInViewport(element)) {
+      element.scrollIntoView({
+        behavior: 'smooth',
+        block: 'nearest',
+        inline: 'start'
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    if (selectedNamespace) {
+      scrollToItem(selectedNamespace);
+    }
+  }, [selectedNamespace, scrollToItem]);
+
+  const renderLongRow = () => (
+    <div 
+      ref={containerRef}
+      className="row flex-nowrap overflow-auto py-1" 
+      style={{ scrollSnapType: 'x mandatory' }}
+    >
+      {namespaces ? (
+        Object.values(namespaces).map((item, index) => (
```

Comment:
Not a huge deal, but a pattern I like over this is:

```tsx
{namespaces && (
 <div> { ... } </div>
 )}
 ```
 Instead of:
 ```tsx
 {namespaces ? (
   <div> { ... } </div>
   ) : null
   }
   ```
   Almost the same, but the first is a bit cleaner